### ISSUE TITLE
Default catalog addition fails for invocation with relative -jar path (and absolute path on Windows)

### ISSUE THREAD

**Issue by gimsieke** [ndw/xmlcalabash1#270]
Lines 222 and 226 in [core/XProcConfiguration.java](https://github.com/ndw/xmlcalabash1/blob/b65e5e3808c4bfd2e10751078bf897b774ea43dc/src/main/java/com/xmlcalabash/core/XProcConfiguration.java#L222) in the Saxon 9.7 branch should be:
```java
catalogs.add(getClass().getResource("/catalog.xml").toString());
```
and
```java
catalogs.add(getClass().getResource("/META-INF/catalog.xml").toString());
```
respectively.

I can create a pull request but I don’t know whether this will work. In order to make it compile, I needed to change `Configuration.softwareEdition` into `Version.softwareEdition`.

But then I could add my top-level catalog as `/META-INF/catalog.xml` and it would work on Cygwin with a Windows Java using both `java -jar $(cygpath -ma xmlcalabash-1.1.19-97.jar)` and `java -jar xmlcalabash-1.1.19-97.jar`. Yay!

Otherwise, it would try to load `jar:file://xmlcalabash-1.1.19-97.jar!/META-INF/catalog.xml` for the relative path or `jar:file://C:/cygwin/home/gerrit/…/xmlcalabash-1.1.19-97.jar!/META-INF/catalog.xml` for the absolute path. The first one doesn’t exist even on Linux and the second one should start with either `jar:file:///C:/` or `jar:file:/C:/`. This string fumbling can be avoided with `getClass().getResource()` which will return the correct jar: URLs in any case.

**Comment by ndw** [comment]
`getClass().getResource()` only returns the first catalog file, not all of the catalog files, so I think I still have to do the loop. And I really don't understand your comment about `Version.softwareEdition`.
Can you please test #271 and tell me if it works for you?

**Comment by gimsieke** [comment]
Re `Configuration.softwareEdition`:

When running javac, I received errors like this one:

```
XProcConfiguration.java:178: error: cannot find symbol
        saxonProcessor = Configuration.softwareEdition.toLowerCase();
```

I could avoid them by adding `import net.sf.saxon.Version;` and changing `Configuration.softwareEdition` to `Version.softwareEdition`. When i look at the API docs, only [`Version`](http://www.saxonica.com/html/documentation9.7/javadoc/net/sf/saxon/Version.html) seems to have a property called `softwareEdition`, not [`Configuration`](http://www.saxonica.com/html/documentation9.7/javadoc/net/sf/saxon/Configuration.html). But what do I know about Java?

**Comment by gimsieke** [comment]
I compiled the core/XProcConfiguration.java of #271, this time without a javac complaint.
After I updated the resulting classes in xmlcalabash-1.1.19-97.jar, my test case still worked, both for the absolute Windows path and for the relative path invocations. Thanks!

**Comment by ndw** [comment]
In `XProcConfiguration.java` in the `saxon97` branch of XML Calabash, line 178 is:

```java
saxonProcessor = Version.softwareEdition.toLowerCase();
```
So I think you must have had the wrong branch checked out or something.

But it sounds like my patch worked for you. I'll apply it and make a new release this evening.

**Comment by gimsieke** [comment]
Ah yes, I cloned the saxon96 branch, apparently.

**Comment by ndw** [comment]
Fixed in 1.1.20.
